src/vehicle.cpp

```diff
@@ -6849,64 +6841,88 @@ long vehicle_part::ammo_capacity() const
 {
     // @todo currently only support fuel tanks and batteries
 
-    if( base.is_magazine() ) {
+    if( base.is_magazine() || base.typeId() == "minireactor" ) {
         return base.ammo_capacity();
     }
 
-    return info().has_flag( VPFLAG_FUEL_TANK ) ? info().size : 0;
+    if( base.is_watertight_container() ) {
+        return base.type->container->contains * std::max( item::find_type( ammo_current() )->stack_size, 1 );
```

Comment:
> I'm not sure if the code for  item::liquid_charges  is correct -

Than it needs to be fixed. It's supposed to encapsulate the volume vs stack size ratio of the different liquid and the caller depend on that being correct.

> Changing  liquid_charges  isn't feasible as much code probably accidentally depends upon this

Callers depend on the *correct* return value.

For example `item::in_its_container` and `item::get_remaining_capacity_for_liquid` us this. If you use a different logic for the tank volume, we get even more problems and confusion.